xclim/generic.py

```diff
@@ -0,0 +1,265 @@
+    else:
+        raise ValueError("mode `{}` should be either 'max' or 'min'".format(mode))
+
+    data = dask.array.apply_along_axis(func, p.get_axis_num('dparams'), p)
+
+    # Create coordinate for the return periods
+    coords = dict(p.coords.items())
+    coords.pop('dparams')
+    coords['return_period'] = t
+
+    # Create dimensions
+    dims = list(p.dims)
+    dims.remove('dparams')
+    dims.insert(0, u'return_period')
+
+    # TODO: add time and time_bnds coordinates (Low will work on this)
+    # time.attrs['climatology'] = 'climatology_bounds'
+    # coords['time'] =
+    # coords['climatology_bounds'] =
+
+    out = xr.DataArray(data=data, coords=coords, dims=dims)
+    out.attrs = p.attrs
+    out.attrs['standard_name'] = '{0} quantiles'.format(dist)
+    out.attrs['long_name'] = '{0} return period values for {1}'.format(dist, getattr(arr, 'standard_name', ''))
+    out.attrs['cell_methods'] = (out.attrs.get('cell_methods', '') + ' dparams: ppf').strip()
+    out.attrs['units'] = arr.attrs.get('units', '')
+    out.attrs['mode'] = mode
+    out.attrs['history'] = out.attrs.get('history', '') + "Compute values corresponding to return periods."
+
+    return out
+
+
+def frequency_analysis(da, mode, t, dist, window=1, freq='YS', **indexer):
+    """Return the value corresponding to a return period.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+      Input data.
+    t : int or sequence
+      Return period. The period depends on the resolution of the input data. If the input array's resolution is
+      yearly, then the return period is in years.
+    dist : str
+      Name of the univariate distribution, such as beta, expon, genextreme, gamma, gumbel_r, lognorm, norm
+      (see scipy.stats).
+    mode : {'min', 'max'}
+      Whether we are looking for a probability of exceedance (high) or a probability of non-exceedance (low).
+    window : int
+      Averaging window length (days).
+    freq : str
+      Resampling frequency. If None, the frequency is assumed annual and adjusted to the seasons. That is,
+      years will start in December if season is 'DJF'.
+    **indexer : {dim: indexer, }, optional
+      Time attribute and values over which to subset the array. For example, use season='DJF' to select winter values,
+      month=1 to select January, or month=[6,7,8] to select summer months. If not indexer is given, all values are
+      considered.
+
+    Returns
+    -------
+    xarray.DataArray
+      An array of values with a 1/t probability of exceedance or non-exceedance when mode is high or low respectively.
+
+    """
+    # Apply rolling average
+    if window > 1:
+        da = da.rolling(time=window, center=False).mean()
+
+    # Assign default resampling frequency if not provided
+    freq = freq or default_freq(**indexer)
+
+    # Extract the time series of min or max over the period
+    sel = select_resample_op(da, op=mode, freq=freq, **indexer).dropna(dim='time')
+
```

Comment:
I assume we drop nans because the scipy fitting function will not accept them, but do we need to check to make sure there is a large enough sample to properly fit the dist in the subsequent fa() call?

Reply:
That's something that should be implemented in the missing method.

Reply:
Opened an issue to account for period selection in missing function.